File: exam-slammr-rest/exam-slammr-rest-user/src/main/java/uk/co/claritysoftware/exam/slammr/rest/user/service/dynamodb/convertor/ZonedDateTimeConverter.java
package uk.co.claritysoftware.exam.slammr.rest.user.service.dynamodb.convertor;

import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter;

public class ZonedDateTimeConverter implements DynamoDBTypeConverter<String, ZonedDateTime> {

    @Override
    public String convert(ZonedDateTime zonedDateTime) {
        return zonedDateTime.format(DateTimeFormatter.ISO_ZONED_DATE_TIME);
    }

    @Override
    public ZonedDateTime unconvert(String stringValue) {
        return ZonedDateTime.parse(stringValue, DateTimeFormatter.ISO_ZONED_DATE_TIME);
    }
}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/persistence/dynamodb/convertor/ZonedDateTimeConverter.kt
package uk.co.claritysoftware.exam.slammr.webapp.persistence.dynamodb.convertor

import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter

class ZonedDateTimeConverter : DynamoDBTypeConverter<String, ZonedDateTime> {

    override fun convert(zonedDateTime: ZonedDateTime): String {
        return zonedDateTime.format(DateTimeFormatter.ISO_ZONED_DATE_TIME)
    }

    override fun unconvert(stringValue: String): ZonedDateTime {
        return ZonedDateTime.parse(stringValue, DateTimeFormatter.ISO_ZONED_DATE_TIME)
    }
}
